src/core/expr/cumsum.cc

```diff
@@ -0,0 +1,127 @@
+#include "expr/fexpr_func.h"
+#include "column/virtual.h"
+#include "expr/eval_context.h"
+#include "expr/workframe.h"
+#include "stype.h"
+#include "python/xargs.h"
+#include "documentation.h"
+#include <numeric>
+#include <vector>
+
+namespace dt {
+namespace expr {
+
+template <typename T>
+class Column_cumsum : public Virtual_ColumnImpl{
+    private:
+      Column acol_;
+
+    public:
+      Column_cumsum(Column&& a)
+        : Virtual_ColumnImpl(a.nrows(), a.stype()),
+          acol_(std::move(a))
+
+    {
+      xassert(acol_.can_be_read_as<T>());
+    }
+
+      ColumnImpl* clone() const override {
+          return new Column_cumsum(Column(acol_));
+      }
+
+      size_t n_children() const noexcept override {
+      return 1;
+    }
+
+    const Column& child(size_t) const override {
+      return acol_;
+    }
+
+
+    bool get_element(size_t i, T* out) const override {
+      T tmp = 0;
+      for (size_t i = 0; i < acol_.nrows(); ++i){
+        T a;
+        bool isvalid = acol_.get_element(i, &a);
+        if (isvalid){
+          tmp = a + tmp;
+          a = tmp;
+          *out = a;
+        }
+        return true;
+      }
+
+      return false;
+     }
```

Comment:
```suggestion
    bool get_element(size_t i, T* out) const override {
      bool out_valid = false;
      for (size_t i = 0; i < acol_.nrows(); ++i){
        T a;
        bool isvalid = acol_.get_element(i, &a);
        if (isvalid){
          out_valid = true;
          *out += a;
        }
      }
      return out_valid;
     }
```

Reply:
In your code the loop stops after the first iteration due to `return true;`. Also, it will return `true` no matter if the cumsum is valid or not. Take a look at my suggestion, I didn't test it but it should give you a rough idea as to what should happen in the `get_element()`. 

Please note, this code is just a sample. In reality it won't be efficient due to the fact that we re-calculate cumsums over and over again. To avoid it, we should do something similar to [qcut implementation](https://github.com/h2oai/datatable/blob/main/src/core/expr/fexpr_qcut.cc), i.e. wrap the column with the `Latent_ColumnImpl` and calculate all the cumsums in one loop producing material column.